sys/include/ztimer.h

```diff
@@ -0,0 +1,489 @@
+ *
+ * @return  Current count on @p clock
+ */
+static inline ztimer_now_t ztimer_now(ztimer_clock_t *clock)
+{
+#if MODULE_ZTIMER_NOW64
+    if (1) {
+#elif MODULE_ZTIMER_EXTEND
+    if (clock->max_value < UINT32_MAX) {
+#else
+    if (0) {
+#endif
+        return _ztimer_now_extend(clock);
+    }
+    else {
+        return clock->ops->now(clock);
+    }
+}
+
+/**
+ * @brief Suspend the calling thread until the time (@p last_wakeup + @p period)
+ *
+ * This function can be used to create periodic wakeups.
+ *
+ * When the function returns, @p last_wakeup is set to
+ * (@p last_wakeup + @p period).
+ *
+ * @c last_wakeup should be set to ztimer_now(@p clock) before first call of the
+ * function.
+ *
+ * If the time (@p last_wakeup + @p period) has already passed, the function
+ * sets @p last_wakeup to @p last_wakeup + @p period and returns immediately.
+ *
+ * @param[in]   clock           ztimer clock to operate on
+ * @param[in]   last_wakeup     base time stamp for the wakeup
+ * @param[in]   period          time in ticks that will be added to @p last_wakeup
+ */
+void ztimer_periodic_wakeup(ztimer_clock_t *clock, uint32_t *last_wakeup,
+                            uint32_t period);
+
+/**
+ * @brief   Put the calling thread to sleep for the specified number of ticks
+ *
+ * @param[in]   clock           ztimer clock to use
+ * @param[in]   duration        duration of sleep, in @p ztimer time units
+ */
+void ztimer_sleep(ztimer_clock_t *clock, uint32_t duration);
+
+/**
+ * @brief Set a timer that wakes up a thread
+ *
+ * This function sets a timer that will wake up a thread when the timer has
+ * expired.
+ *
+ * @param[in] clock         ztimer clock to operate on
+ * @param[in] timer         timer struct to work with.
+ * @param[in] offset        clock ticks from now
+ * @param[in] pid           pid of the thread that will be woken up
+ */
+void ztimer_set_wakeup(ztimer_clock_t *clock, ztimer_t *timer, uint32_t offset,
+                       kernel_pid_t pid);
+
+/**
+ * @brief    Set timeout thread flag after @p timeout
+ *
+ * This function will set THREAD_FLAG_TIMEOUT on the current thread after @p
+ * timeout usec have passed.
+ *
+ * @param[in]   clock           ztimer clock to operate on
+ * @param[in]   timer           timer struct to use
+ * @param[in]   timeout         timeout in ztimer_clock's ticks
+ */
+void ztimer_set_timeout_flag(ztimer_clock_t *clock, ztimer_t *timer,
+                             uint32_t timeout);
+
+/**
+ * @brief   Measure ztimer overhead
+ *
+ * This function can be used to measure the overhead incurred by ztimer.
+ * It will configure a callback to trigger after @p base ticks, then return the
+ * number of ticks that have passed, minus @p base.
+ *
+ * @param[in]   clock   ztimer clock to operate on
+ * @param[in]   base    base interval to use
+ * @return  (time from ztimer_set() until callback) - base
+ */
+uint32_t ztimer_overhead(ztimer_clock_t *clock, uint32_t base);
```

Comment:
Hm, another header? ... If this is important for you, it might be for others, then I'd move this. Lemme know.
But `ztimer_overhead()` can also be used outside of tests.

Reply:
> If this is important for you, it might be for others, then I'd move this. Lemme know.

I wouldn't call it an 'important' matter, but I think it is cleaner (also in the docs) to have it in a separate header as it is not directly relevant for daily usage of ztimer.

Reply:
done